eslint.config.mjs

```diff
@@ -0,0 +1,85 @@
+import eslintConfigSeek from 'eslint-config-seek';
+import importPlugin from 'eslint-plugin-import';
+import { readFileSync } from 'fs';
+import { dirname, join, relative } from 'path';
+import { load as loadYaml } from 'js-yaml';
+import fastGlob from 'fast-glob';
+const { isDynamicPattern, glob } = fastGlob; // eslint-disable-line import-x/no-named-as-default-member -- commonjs module, will move to built in with node 22.
+
+/**
+ * Ignore linting for all files that are gitignored across all workspaces.
+ */
+const rootDir = dirname(import.meta.filename);
+const { packages: workspaces } = loadYaml(
+  readFileSync(join(rootDir, 'pnpm-workspace.yaml'), 'utf8'),
+);
+const gitIgnoresFromWorkspaces = [];
+
+for (const workspace of workspaces) {
+  const pathFromRoot = join(rootDir, workspace);
+  const workspacePaths = isDynamicPattern(pathFromRoot)
+    ? await glob(pathFromRoot, {
+        onlyDirectories: true,
+      })
+    : [pathFromRoot];
+
+  for (const workspacePath of workspacePaths) {
+    try {
+      gitIgnoresFromWorkspaces.push(
```

Comment:
I know this isn't much code, but would it be better to use the [compat utility function](https://github.com/eslint/rewrite/tree/main/packages/compat#including-ignore-files)?

```ts
import { includeIgnoreFile } from "@eslint/compat";
...
includeIgnoreFile(gitignorePath),
```

You'd still need to map over the files to be relative to rootDir:

> Limitation: This works without modification when the ignore file is in the same directory as your config file. If the ignore file is in a different directory, you may need to modify the patterns manually.